File: api/src/main/java/me/kvalbrus/multibans/api/punishment/PunishmentType.java
package me.kvalbrus.multibans.api.punishment;

import lombok.Getter;
import me.kvalbrus.multibans.api.punishment.punishments.Kick;
import me.kvalbrus.multibans.api.punishment.punishments.PermanentlyBan;
import me.kvalbrus.multibans.api.punishment.punishments.PermanentlyBanIp;
import me.kvalbrus.multibans.api.punishment.punishments.PermanentlyChatMute;
import me.kvalbrus.multibans.api.punishment.punishments.TemporaryBan;
import me.kvalbrus.multibans.api.punishment.punishments.TemporaryBanIp;
import me.kvalbrus.multibans.api.punishment.punishments.TemporaryChatMute;
import org.jetbrains.annotations.NotNull;

public enum PunishmentType {

    TEMP_BAN_IP("TEMP_BAN_IP", TemporaryBanIp.class),
    BAN_IP("BAN_IP", PermanentlyBanIp.class),

    TEMP_BAN("TEMP_BAN", TemporaryBan.class),
    BAN("BAN", PermanentlyBan.class),

    TEMP_MUTE("TEMP_MUTE", TemporaryChatMute.class),
    MUTE("MUTE", PermanentlyChatMute.class),

    KICK("KICK", Kick.class);

    @Getter
    private final String prefix;

    @Getter
    private final Class<? extends Punishment> clazz;

    PunishmentType(@NotNull String prefix, @NotNull Class<? extends Punishment> clazz) {
        this.prefix = prefix;
        this.clazz = clazz;
    }
}

File: api/src/main/kotlin/me/kvalbrus/multibans/api/punishment/punishments/PunishmentType.kt
package me.kvalbrus.multibans.api.punishment.punishments

import me.kvalbrus.multibans.api.punishment.Punishment

enum class PunishmentType(val prefix: String, val clazz: Class<out Punishment?>) {
    TEMP_BAN_IP("TEMP_BAN_IP", TemporaryBanIp::class.java),
    BAN_IP("BAN_IP", PermanentlyBanIp::class.java),
    TEMP_BAN("TEMP_BAN", TemporaryBan::class.java),
    BAN("BAN", PermanentlyBan::class.java),
    TEMP_MUTE("TEMP_MUTE", TemporaryChatMute::class.java),
    MUTE("MUTE", PermanentlyChatMute::class.java),
    KICK("KICK", Kick::class.java)
}